diff --git a/.DS_Store b/.DS_Store
index 723e598..ee4d707 100644
Binary files a/.DS_Store and b/.DS_Store differ
diff --git a/CapstoneProject.xcodeproj/project.pbxproj b/CapstoneProject.xcodeproj/project.pbxproj
index ea23b35..35e3dd3 100644
--- a/CapstoneProject.xcodeproj/project.pbxproj
+++ b/CapstoneProject.xcodeproj/project.pbxproj
@@ -9,6 +9,7 @@
 /* Begin PBXBuildFile section */
 		13E20A46289B492500C2CE12 /* ActivityRecView.m in Sources */ = {isa = PBXBuildFile; fileRef = 13E20A45289B492500C2CE12 /* ActivityRecView.m */; };
 		340771E69C21A348911C9217 /* Pods_CapstoneProject.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = FA8B93BEB286AB9F3D1316F3 /* Pods_CapstoneProject.framework */; };
+		4F001BBA28A1B20300AF1FCD /* AXAnimationHelper.m in Sources */ = {isa = PBXBuildFile; fileRef = 4F001BB928A1B20300AF1FCD /* AXAnimationHelper.m */; };
 		764D6C76F584A7E1DD57850C /* Pods_CapstoneProjectUITests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 89366882C845FE22C63CE62D /* Pods_CapstoneProjectUITests.framework */; };
 		EC576E09A084256B4E300673 /* Pods_CapstoneProjectTests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 66950B703E5F785384E3108A /* Pods_CapstoneProjectTests.framework */; };
 		F576C0DE287F46E300C93523 /* ActivityRecViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = F576C0DD287F46E300C93523 /* ActivityRecViewController.m */; };
@@ -57,6 +58,8 @@
 		212AFF4A3BDE66FD65861084 /* Pods-CapstoneProjectTests.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-CapstoneProjectTests.debug.xcconfig"; path = "Pods/Target Support Files/Pods-CapstoneProjectTests/Pods-CapstoneProjectTests.debug.xcconfig"; sourceTree = "<group>"; };
 		3BF5F8F5A7A1B8A63FA53C7F /* Pods-CapstoneProjectUITests.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-CapstoneProjectUITests.debug.xcconfig"; path = "Pods/Target Support Files/Pods-CapstoneProjectUITests/Pods-CapstoneProjectUITests.debug.xcconfig"; sourceTree = "<group>"; };
 		42A3D1D3588070588CAA87E2 /* Pods-CapstoneProjectTests.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-CapstoneProjectTests.release.xcconfig"; path = "Pods/Target Support Files/Pods-CapstoneProjectTests/Pods-CapstoneProjectTests.release.xcconfig"; sourceTree = "<group>"; };
+		4F001BB828A1B20300AF1FCD /* AXAnimationHelper.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = AXAnimationHelper.h; sourceTree = "<group>"; };
+		4F001BB928A1B20300AF1FCD /* AXAnimationHelper.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = AXAnimationHelper.m; sourceTree = "<group>"; };
 		5B23CEE57A0409C7FC50A72A /* Pods-CapstoneProject.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-CapstoneProject.release.xcconfig"; path = "Pods/Target Support Files/Pods-CapstoneProject/Pods-CapstoneProject.release.xcconfig"; sourceTree = "<group>"; };
 		66950B703E5F785384E3108A /* Pods_CapstoneProjectTests.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_CapstoneProjectTests.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		89366882C845FE22C63CE62D /* Pods_CapstoneProjectUITests.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_CapstoneProjectUITests.framework; sourceTree = BUILT_PRODUCTS_DIR; };
@@ -132,6 +135,15 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
+		4F001BB728A1B1DB00AF1FCD /* Animations */ = {
+			isa = PBXGroup;
+			children = (
+				4F001BB828A1B20300AF1FCD /* AXAnimationHelper.h */,
+				4F001BB928A1B20300AF1FCD /* AXAnimationHelper.m */,
+			);
+			path = Animations;
+			sourceTree = "<group>";
+		};
 		5709A0E69B1311F21D61DCCE /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
@@ -218,6 +230,7 @@
 		F5D03197287C8F520065FDF3 /* CapstoneProject */ = {
 			isa = PBXGroup;
 			children = (
+				4F001BB728A1B1DB00AF1FCD /* Animations */,
 				F5D03198287C8F520065FDF3 /* AppDelegate.h */,
 				F5D03199287C8F520065FDF3 /* AppDelegate.m */,
 				F5D031A4287C8F530065FDF3 /* Assets.xcassets */,
@@ -466,6 +479,7 @@
 			files = (
 				F5D0319A287C8F520065FDF3 /* AppDelegate.m in Sources */,
 				F5D031AB287C8F530065FDF3 /* main.m in Sources */,
+				4F001BBA28A1B20300AF1FCD /* AXAnimationHelper.m in Sources */,
 				F576C0DE287F46E300C93523 /* ActivityRecViewController.m in Sources */,
 				F576C0FC288731A100C93523 /* ActivityRecommendationTableViewCell.m in Sources */,
 				F576C102289B1EA800C93523 /* ProfileViewController.m in Sources */,
diff --git a/CapstoneProject.xcworkspace/xcuserdata/apoorvac.xcuserdatad/UserInterfaceState.xcuserstate b/CapstoneProject.xcworkspace/xcuserdata/apoorvac.xcuserdatad/UserInterfaceState.xcuserstate
index f2777be..9afa678 100644
Binary files a/CapstoneProject.xcworkspace/xcuserdata/apoorvac.xcuserdatad/UserInterfaceState.xcuserstate and b/CapstoneProject.xcworkspace/xcuserdata/apoorvac.xcuserdatad/UserInterfaceState.xcuserstate differ
diff --git a/CapstoneProject/Animations/AXAnimationHelper.h b/CapstoneProject/Animations/AXAnimationHelper.h
index 874a98a..9766a65 100644
--- a/CapstoneProject/Animations/AXAnimationHelper.h
+++ b/CapstoneProject/Animations/AXAnimationHelper.h
@@ -9,8 +9,17 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface HappyAnimationDestination : NSObject
+
+@property (nonatomic) int x;
+@property (nonatomic) int y;
+
+@end
+
 @interface AXAnimationHelper : NSObject
 
++ (HappyAnimationDestination *)generateHappyDestination;
+
 @end
 
 NS_ASSUME_NONNULL_END
diff --git a/CapstoneProject/Animations/AXAnimationHelper.m b/CapstoneProject/Animations/AXAnimationHelper.m
index d7db79f..65f1d5e 100644
--- a/CapstoneProject/Animations/AXAnimationHelper.m
+++ b/CapstoneProject/Animations/AXAnimationHelper.m
@@ -7,6 +7,25 @@
 
 #import "AXAnimationHelper.h"
 
+@implementation HappyAnimationDestination
+
+@end
+
+
 @implementation AXAnimationHelper
 
++ (HappyAnimationDestination *)generateHappyDestination {
+    HappyAnimationDestination *destination = [HappyAnimationDestination new];
+    
+    if (arc4random_uniform(2) == 0) {
+        destination.x = -arc4random_uniform(100);
+        destination.y = arc4random_uniform(500);
+    } else {
+        destination.x = arc4random_uniform(500);
+        destination.y = -arc4random_uniform(100);
+    }
+    
+    return destination;
+}
+
 @end
diff --git a/CapstoneProject/Models/MoodDetection.m b/CapstoneProject/Models/MoodDetection.m
index 88260bf..52a4ece 100644
--- a/CapstoneProject/Models/MoodDetection.m
+++ b/CapstoneProject/Models/MoodDetection.m
@@ -53,8 +53,10 @@
 //    NSLog(@"hello");
 //    NSLog(@"%@", self.moodAPIResponse);
     
-    self.firstMood = @"Sad";
-    self.secondMood = @"Happy";
+//    self.firstMood = @"Sad";
+//    self.secondMood = @"Happy";
+    self.firstMood = @"Happy";
+    self.secondMood = @"Sad";
     self.thirdMood = @"Angry";
     
     // figures out the users mood based on what the api returns
diff --git a/CapstoneProject/ViewControllers/ActivityRecViewController.m b/CapstoneProject/ViewControllers/ActivityRecViewController.m
index 4e11013..c67e766 100644
--- a/CapstoneProject/ViewControllers/ActivityRecViewController.m
+++ b/CapstoneProject/ViewControllers/ActivityRecViewController.m
@@ -11,6 +11,7 @@
 #import "MoodDetection.h"
 #import "ChangeEmotionViewController.h"
 #import "DisplayActivityViewController.h"
+#import "AXAnimationHelper.h"
 #import <Parse/Parse.h>
 
 @interface ActivityRecViewController ()
@@ -56,7 +57,8 @@
 
 -(void) loadAnimations {
     if([self.firstMood isEqualToString:@"Happy"]) {
-        for(int i = 0; i < 100; i++){
+        for(int i = 0; i < 50; i++){
+            [CATransaction begin];
             UIImageView *careImage = [[UIImageView alloc] init];
             
             CGRect imageframe = careImage.frame;
@@ -70,18 +72,37 @@
             [self.view addSubview:careImage];
             careImage.layer.zPosition = MAXFLOAT;
             
-            NSArray *pathArray = @[[NSValue valueWithCGPoint:CGPointMake([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height)], [NSValue valueWithCGPoint:CGPointMake((int)arc4random_uniform(500), (int)arc4random_uniform(950))]];
+            CAMediaTimingFunctionName timeFunction = kCAMediaTimingFunctionLinear;
+            if (arc4random_uniform(2) == 0) {
+                timeFunction = kCAMediaTimingFunctionEaseIn;
+            } else {
+                timeFunction = kCAMediaTimingFunctionEaseInEaseOut;
+            }
+            
+            HappyAnimationDestination *destination = [AXAnimationHelper generateHappyDestination];
+            
+            NSArray *pathArray = @[[NSValue valueWithCGPoint:CGPointMake([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height)], [NSValue valueWithCGPoint:CGPointMake(destination.x, destination.y)]];
             CAKeyframeAnimation *pathAnimation = [CAKeyframeAnimation animationWithKeyPath:@"position"];
             pathAnimation.values = pathArray;
-            pathAnimation.duration = 2.5;
-            [careImage.layer addAnimation:pathAnimation forKey:@"position"];
+            pathAnimation.duration = arc4random_uniform(3) + 2;
+            [pathAnimation setRemovedOnCompletion:YES];
+            [pathAnimation setTimingFunction:[CAMediaTimingFunction functionWithName:timeFunction]];
             
             CABasicAnimation *flash = [CABasicAnimation animationWithKeyPath:@"opacity"];
             flash.fromValue = [NSNumber numberWithFloat: (float)arc4random_uniform(100)];
             flash.toValue = [NSNumber numberWithFloat: 0.0];
             flash.duration = (float)arc4random_uniform(5);
             flash.autoreverses = YES;
+            [flash setRemovedOnCompletion:YES];
+            
+            [CATransaction setCompletionBlock:^{
+                [careImage removeFromSuperview];
+            }];
+            
             [careImage.layer addAnimation:flash forKey:@"flashAnimation"];
+            [careImage.layer addAnimation:pathAnimation forKey:@"position"];
+
+            [CATransaction commit];
         }
     } else if([self.firstMood isEqualToString:@"Sad"]) {
         [self sadAnimation];
@@ -264,6 +285,8 @@
     
 }
 
+#pragma mark - UITableViewDelegate/UITableViewDataSource
+
 - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
     ActivityRecommendationTableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"activityCell" forIndexPath:indexPath];
     
